include/SimCore/PluginFactory.h

```diff
@@ -0,0 +1,211 @@
+#ifndef SIMCORE_PLUGINFACTORY_H
```

Comment:
If this manages the generators, you should name it something that hints at that.  PluginFactory is really broad.  I would name it something like "GeneratorFactory"

Reply:
I named it "PluginFactory" becuase it handles generators, biasing operators, and user actions.